src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java

```diff
@@ -0,0 +1,356 @@
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001", "8d6cc5cf-c973-11eb-aaaa-100000000001");
+
+        SearchRequest searchRequest = SearchRequest.builder()
+            .cftTaskStates(List.of(CFTTaskState.ASSIGNED))
+            .jurisdictions(List.of("WA"))
+            .locations(List.of("765324"))
+            .roleCategories(List.of(RoleCategory.JUDICIAL))
+            .workTypes(List.of("hearing_work"))
+            .sortingParameters(List.of(new SortingParameter(SortField.CASE_NAME_CAMEL_CASE, SortOrder.ASCENDANT)))
+            .build();
+
+
+        GetTasksResponse<Task> response = cftTaskDatabaseService.searchForTasks(searchRequest,
+            accessControlResponse, false);
+        assertEquals(1, response.getTotalRecords());
+        Assertions.assertThat(response.getTasks())
+            .hasSize(1)
+            .flatExtracting(Task::getId, Task::getCaseId)
+            .containsExactly(
+                newArrayList(
+                    "8d6cc5cf-c973-11eb-aaaa-000000000001", "1623278362400001"
+                ).toArray()
+            );
+    }
+
+    @Test
+    void should_return_task_list_and_count_when_filter_task_by_role_category() {
+        List<RoleAssignment> roleAssignments = roleAssignmentsWithGrantTypeStandard();
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001", "8d6cc5cf-c973-11eb-aaaa-200000000001");
+
+        SearchRequest searchRequest = SearchRequest.builder()
+            .cftTaskStates(List.of(CFTTaskState.ASSIGNED))
+            .jurisdictions(List.of("WA", "IA"))
+            .locations(List.of("765324"))
+            .roleCategories(List.of(RoleCategory.JUDICIAL))
+            .workTypes(List.of("hearing_work"))
+            .sortingParameters(List.of(new SortingParameter(SortField.CASE_NAME_CAMEL_CASE, SortOrder.ASCENDANT)))
+            .build();
+
+
+        GetTasksResponse<Task> response = cftTaskDatabaseService.searchForTasks(searchRequest,
+            accessControlResponse, false);
+        assertEquals(1, response.getTotalRecords());
+        Assertions.assertThat(response.getTasks())
+            .hasSize(1)
+            .flatExtracting(Task::getId, Task::getCaseId)
+            .containsExactly(
+                newArrayList(
+                    "8d6cc5cf-c973-11eb-aaaa-000000000001", "1623278362400001"
+                ).toArray()
+            );
+    }
+
+    @Test
+    void should_return_task_list_and_count_when_filter_task_by_location() {
+        List<RoleAssignment> roleAssignments = roleAssignmentsWithGrantTypeStandard();
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001", "8d6cc5cf-c973-11eb-aaaa-300000000001");
```

Comment:
same id's indexRecord?

Reply:
This is now addressed in RWA-2215 and all the records are indexed there.